packages/grafana-data/src/transformations/transformers/convertFieldType.ts

```diff
@@ -142,9 +142,16 @@ function fieldToNumberField(field: Field): Field {
   const numValues = field.values.toArray().slice();
 
   for (let n = 0; n < numValues.length; n++) {
-    // some numbers returned from datasources have commas
-    // strip the commas, coerce the string to a number
-    const number = +numValues[n].replace(/,/g, '');
+    let toBeConverted = numValues[n];
+
+    if (typeof toBeConverted === 'string') {
+      // some numbers returned from datasources have commas
```

Comment:
feels like this can be outside this hot loop. isnt this uniform/known for the whole array?

Reply:
Good point! As far as I know, the array should be consistent. I can do the check on the array

Reply:
I can check `field.type === 'string'` or I can check `field.some(v => typeof v === 'string')`

Is there somewhere in the code that all dataframe fields are checked for a type? Is this consistent across all datasources? I'm leaning towards the `some` check.

Reply:
:+1: